src/ui/api/member.go

```diff
@@ -158,12 +161,26 @@ func (pma *ProjectMemberAPI) Post() {
 
 	var req memberReq
 	pma.DecodeJSONReq(&req)
-	username := req.Username
+	username := strings.TrimSpace(req.Username)
 	userID := checkUserExists(username)
 	if userID <= 0 {
-		log.Warningf("User does not exist, user name: %s", username)
-		pma.RenderError(http.StatusNotFound, "User does not exist")
-		return
+		//check current authorization mode
+		authMode, err := config.AuthMode()
+		if err != nil || authMode != "ldap_auth" {
+			log.Warningf("User does not exist, user name: %s", username)
+			pma.RenderError(http.StatusNotFound, "User does not exist")
+			return
+		}
+
+		//search and import user
+		newUserID, err := ldapUtils.SearchAndImportUser(username)
+		if err == nil || newUserID > 0 {
```

Comment:
if err != nil{
  // log error and return 500
}

//else
if newUserID == 0{
  // not found return 404
}

// otherwise continue

Reply:
Updated